test/test_jit.py

```diff
@@ -4125,6 +4125,43 @@ def test_invalid_list_equality():
             RuntimeError,
             "bool value of Tensor")
 
+    def test_list_sort(self):
+        template = dedent('''
+        def func():
+            li = {list_create}
+            li.sort()
+            return li
+        ''')
+
+        lists = ["[]", "[1, 3, 2]", "[True, False, True]", "[1.2, .2, 3.2]",
+                 "[torch.tensor(1.0), torch.tensor(0.2), torch.tensor(0.5)]",
+                 "[torch.tensor(5), torch.tensor(-2), torch.tensor(4)]"]
+        for li in lists:
+            code = template.format(list_create=li)
+            scope = {}
+            exec(code, globals(), scope)
```

Comment:
Let's avoid metaprogramming tests too much; I think the test would be clearer if you just unrolled all these lists and used the regular `checkScript` functionality.